File: Office-Hours/Excel-Java/src/main/java/edu/odu/cs/cs350/examples/Driver.java
package edu.odu.cs.cs350.examples;

import java.util.List;
import java.util.Vector;
import java.util.Map;
import java.util.HashMap;

import java.io.IOException;
import java.io.FileNotFoundException;

public class Driver
{
    /**
     * Generate a small roster for demonstration.
     */
    public static Roster generateRoster()
    {
        Student john  = new Student("John");
        Student tom   = new Student("Tom");
        Student jay   = new Student("Jay");
        Student oscar = new Student("Oscar");

        Student[] allStudents = {john, tom, jay, oscar};

        Roster cs330 = new Roster(4, "CS 330");

        for (Student s : allStudents) {
            final boolean enrollResult = cs330.enroll(s);
        }

        return cs330;
    }

    public static void main(String... args)
        throws IOException, FileNotFoundException
    {
        final Roster roster = generateRoster();

        ExampleExcelWriter writer = new ExampleExcelWriter(roster);

        writer.demoSimpleTable();
    }
}

File: Office-Hours/Excel-Python/driver.py
import sys
from typing import TextIO

from examples.roster import Roster
from examples.student import Student
from simple_excel_writer import *


def generate_roster() -> Roster:

    john = Student("John")
    tom = Student("Tom")
    jay = Student("Jay")
    oscar = Student("Oscar")

    all_students = [john, tom, jay, oscar]

    cs330 = Roster(4, "CS 330")

    for stu in all_students:
        _ = cs330.enroll(stu)

    return cs330


def main() -> None:
    roster = generate_roster()

    demo_simple_table()
    demo_roster_table(roster, destination="report_students.xlsx")


if __name__ == "__main__":
    main()
